capstone/src/test/java/com/google/sps/LessonTest.java

```diff
@@ -28,4 +49,150 @@ public void setUp() {
+
+    @Test(expected = NullPointerException.class)
+    public void deserializeJsonMissingFieldTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, Lesson.TYPE_FORM);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Form.EDIT_URL_PROPERTY_KEY, LessonTest.TEST_EDIT_URL);
+        jsonObject.addProperty(Form.URL_PROPERTY_KEY, LessonTest.TEST_URL);
+
+        Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void deserializeJsonMissingTypeTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TITLE_PROPERTY_KEY, LessonTest.TEST_TITLE);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Form.EDIT_URL_PROPERTY_KEY, LessonTest.TEST_EDIT_URL);
+        jsonObject.addProperty(Form.URL_PROPERTY_KEY, LessonTest.TEST_URL);
+
+        Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void deserializeJsonWrongTypeTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, "MalformedForm");
+        jsonObject.addProperty(Lesson.TITLE_PROPERTY_KEY, LessonTest.TEST_TITLE);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Form.EDIT_URL_PROPERTY_KEY, LessonTest.TEST_EDIT_URL);
+        jsonObject.addProperty(Form.URL_PROPERTY_KEY, LessonTest.TEST_URL);
+
+        Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+    }
+
+    @Test
+    public void deserializeJsonExtraFieldTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, Lesson.TYPE_FORM);
+        jsonObject.addProperty(Lesson.TITLE_PROPERTY_KEY, LessonTest.TEST_TITLE);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Form.EDIT_URL_PROPERTY_KEY, LessonTest.TEST_EDIT_URL);
+        jsonObject.addProperty(Form.URL_PROPERTY_KEY, LessonTest.TEST_URL);
+        jsonObject.addProperty("Random Field", "Random Value");
+
+        Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+
+        Assert.assertEquals(lesson.getType(), Lesson.TYPE_FORM);
+        Assert.assertEquals(lesson.getTitle(), LessonTest.TEST_TITLE);
+        Assert.assertEquals(lesson.getDescription(), LessonTest.TEST_DESCRIPTION);
+        Assert.assertEquals(lesson.getIsDraft(), false);
+        Assert.assertTrue(lesson instanceof Form);
+        Form form = (Form) lesson; 
+        Assert.assertEquals(form.getEditUrl(), LessonTest.TEST_EDIT_URL);
+        Assert.assertEquals(form.getUrl(), LessonTest.TEST_URL);
+    }
```

Comment:
Decided to leave the other tests for each type since I didn't want to let this PR be big. I could add it in a future PR along with servlet testing (I got it to work I think).

Reply:
Wrong area to write this.